axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/InvoiceLineSupplychainService.java

```diff
@@ -82,48 +87,91 @@ public Unit getUnit(Product product, boolean isPurchase) {
+  public Map<String, Object> getDiscount(Invoice invoice, InvoiceLine invoiceLine, BigDecimal price)
+      throws AxelorException {
 
-    PriceList priceList = invoice.getPriceList();
-    BigDecimal discountAmount = BigDecimal.ZERO;
-    int computeMethodDiscountSelect =
-        appAccountService.getAppBase().getComputeMethodDiscountSelect();
+    Map<String, Object> discounts = new HashMap<>();
 
-    Map<String, Object> discounts = super.getDiscount(invoice, invoiceLine, price);
+    if (invoice.getOperationTypeSelect() < InvoiceRepository.OPERATION_TYPE_CLIENT_SALE) {
+      Map<String, Object> catalogInfo = this.updateInfoFromCatalog(invoice, invoiceLine);
 
-    if (priceList != null) {
-      discountAmount = (BigDecimal) discounts.get("discountAmount");
+      if (catalogInfo != null) {
+        if (catalogInfo.get("price") != null) {
+          price = (BigDecimal) catalogInfo.get("price");
+        }
+        discounts.put("productName", catalogInfo.get("productName"));
+      }
     }
 
-    if (invoice.getOperationTypeSelect() < InvoiceRepository.OPERATION_TYPE_CLIENT_SALE
-        && discountAmount.compareTo(BigDecimal.ZERO) == 0) {
-      List<SupplierCatalog> supplierCatalogList = invoiceLine.getProduct().getSupplierCatalogList();
-      if (supplierCatalogList != null && !supplierCatalogList.isEmpty()) {
-        SupplierCatalog supplierCatalog =
-            Beans.get(SupplierCatalogRepository.class)
-                .all()
-                .filter(
-                    "self.product = ?1 AND self.minQty <= ?2 AND self.supplierPartner = ?3 ORDER BY self.minQty DESC",
-                    invoiceLine.getProduct(),
-                    invoiceLine.getQty(),
-                    invoice.getPartner())
-                .fetchOne();
-        if (supplierCatalog != null) {
-
-          discounts = purchaseProductService.getDiscountsFromCatalog(supplierCatalog, price);
-
-          if (computeMethodDiscountSelect != AppBaseRepository.DISCOUNT_SEPARATE) {
-            discounts.put(
-                "price",
-                priceListService.computeDiscount(
-                    price,
-                    (int) discounts.get("discountTypeSelect"),
-                    (BigDecimal) discounts.get("discountAmount")));
-          }
+    discounts.putAll(super.getDiscount(invoice, invoiceLine, price));
+
+    return discounts;
+  }
+
+  private Map<String, Object> updateInfoFromCatalog(Invoice invoice, InvoiceLine invoiceLine)
+      throws AxelorException {
+
+    Map<String, Object> info = null;
+
+    List<SupplierCatalog> supplierCatalogList = invoiceLine.getProduct().getSupplierCatalogList();
+    if (supplierCatalogList != null && !supplierCatalogList.isEmpty()) {
+      SupplierCatalog supplierCatalog =
+          Beans.get(SupplierCatalogRepository.class)
+              .all()
+              .filter(
+                  "self.product = ?1 AND self.minQty <= ?2 AND self.supplierPartner = ?3 ORDER BY self.minQty DESC",
```

Comment:
These process are already done in Purchase module, so better to use it in Supplychain too instead of duplicate the code.